server/controller/blog.go

```diff
@@ -17,15 +18,66 @@ import (
 // @Produce  json
 // @Param account body model.Blog true "Add blog"
 // @Success 200 {object} model.Blog
-// @Router /api/blogs [post]
+// @Router /api/blogs/publish [post]
 func (c *Controller) PublishBlog(ctx *gin.Context) {
 	b := model.Blog{}
 	if err := ctx.ShouldBindJSON(&b); err != nil {
-		httputil.NewError(ctx, http.StatusBadRequest, err)
+		ctx.AbortWithStatusJSON(http.StatusBadRequest, SetResponseReason("内容含非法字符，请检查"))
 		return
 	}
-	// to be implemented: set blog's author due to cookie infomation
+	// set blog's author due to cookie information
+	info, ext := mo2utils.GetUserInfo(ctx)
```

Comment:
这一步之后需要检查用户roles是否包含User
```suggestion
	info, ext := mo2utils.GetUserInfo(ctx)
	if !mo2utils.Contains(info.Roles, model.OrdinaryUser){
	    ctx.AbortWithStatusJSON(http.StatusForbidden, SetResponseReason("Unauthorized!"))
	}
```这是最简单的写法  
另外我们应该讨论一下是否要把身份验证统一加到中间件里

Reply:
还有我们可以讨论一下是否要统一将身份验证移到中间件里

Reply:
好的，等讨论后统一修改